Packs/FortiGate/Integrations/FortiGate/FortiGate.py

```diff
@@ -103,1209 +2141,3696 @@ def create_addr_string(list_of_addr_data_dicts):
+        negate_source_address=negate_source_address,
+        negate_destination_address=negate_destination_address,
+        services=services,
+        negate_service=negate_service,
+        action="deny" if action == "block" else action,
+        status=status,
+        log_traffic=log_traffic,
+        schedule=schedule,
+        nat=nat,
+        is_address_v4=is_address_v4,
+    )
+
+    outputs = {
+        "Name": name,
+        "Description": comment,
+        "Status": status,
+        "Service": service,
+        "Action": action,
+        "Log": log_traffic,
+        "Source": {
+            "Interface": source_interface,
+            "Address": policy_addr_array_from_arg(source_address),
+            "Address6": policy_addr_array_from_arg(source_address6),
+        },
+        "Destination": {
+            "Interface": destination_interface,
+            "Address": policy_addr_array_from_arg(destination_address),
+            "Address6": policy_addr_array_from_arg(destination_address6),
+        },
+        "NAT": nat,
+    }
+    readable_output = f"## The firewall policy '{name}' was successfully created."
+
+    return CommandResults(
+        outputs_prefix=POLICY_CONTEXT,
+        outputs_key_field="Name",
+        outputs=outputs,
+        readable_output=readable_output,
+        raw_response=response,
+    )
+
+
+@logger
+def update_firewall_policy_command(client: Client, args: dict[str, Any]) -> CommandResults:
+    """Update firewall policies.
+
+    Args:
+        client (Client): Session to Fortigate to run API requests.
+        args (dict[str, Any]): Arguments passed down by the CLI to configure the request.
+
+    Raises:
+        DemistoException: If 'keep_original_data' is 'True', but 'add_or_remove' wasn't provided.
+
+    Returns:
+        CommandResults: Outputs of the command that represent an entry in warroom.
+    """
+    vdom = args.get("vdom", DEFAULT_VDOM)
+    id_ = args.get("policyID", "")
+    field = args.get("field", "")
+    value = args.get("value", "")
+    keep_original_data = argToBoolean(args.get("keep_original_data", False))
+    add_or_remove = args.get("add_or_remove")
+
+    if keep_original_data and not add_or_remove:
+        raise DemistoException("If 'keep_original_data' is set to True, 'add_or_remove' must also be set.")
+
+    field_to_api_key = {
+        "source_interface": "srcintf",
+        "destination_interface": "dstintf",
+        "description": "comments",
+        "source": "srcaddr",
+        "destination": "dstaddr",
+        "source6": "srcaddr6",
+        "destination6": "dstaddr6",
+        "log": "logtraffic",
+        "negate_source": "srcaddr-negate",
+        "negate_destination": "dstaddr-negate",
+        "negate_source6": "srcaddr6-negate",
+        "negate_destination6": "dstaddr6-negate",
+        "negate_service": "service-negate",
+    }
+    api_key = field_to_api_key.get(field, field)
+
+    if field in {
```

Comment:
Please extract the 2 nested 'if's to a separate function with a proper doc_string explaining the purpose of this code.

Reply:
I'm not too fond of nesting, but isn't the logic already broken apart to multiple functions enough already? Also this will mean I'll have to pass a `Callable` for `client.list_firewall_policies` plus ~5 other arguments.